From 2469980899d09cdab121f3f1daa36a5419b00103 Mon Sep 17 00:00:00 2001
From: Koan-Sin Tan <freedom.tan@mediatek.com>
Date: Wed, 26 Oct 2016 13:27:32 +0800
Subject: [PATCH 2/3] add device node for WDT

Add MediaTek watchdog timer (WDT) device for MT6797, so that we
don't need to disable the WDT before the Linux kernel, such as
in the LK, the bootloader.
---
 arch/arm64/boot/dts/mediatek/mt6797.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/mt6797.dtsi b/arch/arm64/boot/dts/mediatek/mt6797.dtsi
index 1d42ad4..fd2d6d9 100644
--- a/arch/arm64/boot/dts/mediatek/mt6797.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt6797.dtsi
@@ -175,6 +175,12 @@
 		infracfg = <&infrasys>;
 	};
 
+	watchdog: watchdog@10007000 {
+		compatible = "mediatek,mt6797-wdt",
+			     "mediatek,mt6589-wdt";
+		reg = <0 0x10007000 0 0x100>;
+	};
+
 	apmixedsys: apmixed@1000c000 {
 		compatible = "mediatek,mt6797-apmixedsys";
 		reg = <0 0x1000c000 0 0x1000>;
-- 
2.7.4

